### PR TITLE
build: fix container image targets

### PR DESCRIPTION
**Pull request by rchincha**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -338,8 +338,8 @@ run-container:
 	${CONTAINER_RUNTIME} run --rm --security-opt label=disable -v $$(pwd):/go/src/github.com/project-zot/zot \
 		zot-build:latest
 
-.PHONY: binary-stacker
-binary-stacker: $(STACKER)
+.PHONY: oci-image
+oci-image: $(STACKER)
```
[the remark this answers is not in the document]
Comment (adodon2go) [reply Makefile:342]: Do we need binary-container target (see above)? 
Seams that it is not really used (in CI for example).

[the remark this answers is not in the document]
Comment (rchincha) [reply Makefile:342]: Fixed. Yes not really used - this was from early days ...

File: build/Dockerfile-conformance [hunk 2]
```diff
@@ -2,7 +2,11 @@
 # Stage 1: Install certs, build binary, create default config file
 # ---
 FROM ghcr.io/project-zot/golang:1.20 AS builder
```
Comment (andaaron) [inline build/Dockerfile-conformance:4]: Don't we need `--platform=$BUILDPLATFORM` here, and in the other Dockerfiles?

Comment (rchincha) [reply build/Dockerfile-conformance]: good catch! fixed

File: Makefile [hunk 3]
```diff
@@ -348,9 +348,9 @@ binary-stacker: $(STACKER)
 		--substitute RELEASE_TAG=$(RELEASE_TAG) \
 		--substitute REPO_NAME=zot-$(OS)-$(ARCH)
 
-.PHONY: image
-image:
-	${CONTAINER_RUNTIME} build ${BUILD_ARGS} -f build/Dockerfile -t zot:latest .
+.PHONY: docker-image
+docker-image:
+	${CONTAINER_RUNTIME} buildx build --platform $(OS)/$(ARCH) -f build/Dockerfile .
```
Comment (adodon2go) [inline Makefile:353]: Does this work with podman? Locally (on my side) this target fails.
Should we use docker directly?

File: Makefile [hunk 4]
```diff
@@ -338,8 +338,8 @@ run-container:
 	${CONTAINER_RUNTIME} run --rm --security-opt label=disable -v $$(pwd):/go/src/github.com/project-zot/zot \
 		zot-build:latest
 
-.PHONY: binary-stacker
-binary-stacker: $(STACKER)
+.PHONY: oci-image
```
Comment (adodon2go) [inline Makefile:341]: This target can be run only on linux. In fact $(STACKER) depends on check-linux target.

> aiandu@aiandus-MacBook-Pro zot % git diff .
> diff --git a/Makefile b/Makefile
> index f2ef779..d108907 100644
> --- a/Makefile
> +++ b/Makefile
> @@ -421,7 +421,7 @@ fuzz-all:
>         bash test/scripts/fuzzAll.sh ${fuzztime}; \
>         rm -rf pkg/storage/testdata; \
>  
> -$(STACKER):
> +$(STACKER): check-linux
>         mkdir -p $(TOOLSDIR)/bin; \
>         curl -fsSL https://github.com/project-stacker/stacker/releases/latest/download/stacker -o $@; \
>         chmod +x $@

File: Makefile (hunk 3 above)
Comment (rchincha) [reply Makefile:353]: $ make CONTAINER_RUNTIME=podman docker-image

File: Makefile (hunk 4 above)
Comment (rchincha) [reply Makefile:341]: Correct, `stacker` is currently supported only on host platform linux/amd64. Target will depend on the toolchain used inside stacker.

File: build/Dockerfile-zb [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
+
+ARG TARGETOS
+ARG TARGETARCH
 ARG COMMIT
-ARG OS
-ARG ARCH
+
 RUN mkdir -p /go/src/github.com/project-zot/zot
 WORKDIR /go/src/github.com/project-zot/zot
 COPY . .
```
Comment (andaaron) [inline build/Dockerfile-zb:12]: The make command below needs an update the replace the old OS and ARCH variables with the new ones.

File: Makefile (hunk 4 above)
Comment (rchincha) [reply Makefile:357]: updated the Makefile rule

**Comment by rchincha** [comment]
Added a nightly test just in case

File: Makefile (hunk 3 above)
Comment (rchincha) [reply Makefile:369]: https://github.com/rchincha/zot/actions/runs/6396870937/job/17363595069

podman works

**Review by adodon2go — APPROVED** [review]

**Pull request merged**